docs/blog/vitest-3.2.md

```diff
@@ -0,0 +1,199 @@
+
+```ts
+import { locators } from '@vitest/browser/context'
+import type { Locator } from '@vitest/browser/context'
+
+locators.extend({
+  getByCommentsCount(this: Locator, count: number) {
+    return this.getByRole('comment')
+      .and(this.getByText(`${count} comments`))
+  },
+})
+```
+
+Having access to context also allows you to call regular methods of the locator to define a custom user event:
+
+```ts
+import { locators, page } from '@vitest/browser/context'
+import type { Locator } from '@vitest/browser/context'
+
+locators.extend({
+  clickAndFill(this: Locator, text: string) {
+    await this.click()
+    await this.fill(text)
+  },
+})
+
+await page.getByRole('textbox').clickAndFill('Hello World')
+```
+
+Please, refer to the [`locators.extend` API](/guide/browser/locators#custom-locators) for more information.
+
+## Explicit Resource Management in `vi.spyOn` and `vi.fn`
+
+In environments that support [Explicit Resource Management](https://github.com/tc39/proposal-explicit-resource-management), you can use `using` instead of `const` to automatically call `mockRestore` on any mocked function when the containing block is exited. This is especially useful for spied methods:
+
+```ts
+it('calls console.log', () => {
+  using spy = vi.spyOn(console, 'log').mockImplementation(() => {})
+  debug('message')
+  expect(spy).toHaveBeenCalled()
+})
+
+// console.log is restored here
+```
+
+## Test `signal` API
+
+Vitest now provides an [`AbortSignal`](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal) object to the test body. You can use it to stop any resource that supports this Web API.
+
+The signal is aborted when test times out, another test fails and [`--bail` flag](/config/#bail) is set to a non-zero value, or the user presses Ctrl+C in the terminal.
+
+For example, you can stop a `fetch` request when tests are interrupted:
+
+```ts
+it('stop request when test times out', async ({ signal }) => {
+  await fetch('/heavy-resource', { signal })
+}, 2000)
+```
+
+## Coverage V8 AST-aware remapping
+
+Vitest now uses `ast-v8-to-istanbul` package developed by one of the Vitest maintainers, [AriPerkkio](https://github.com/AriPerkkio). This brings v8 coverage report in line with istanbul, but has a better performance! Enable this feature by setting [`coverage.experimentalAstAwareRemapping`](/config/#coverage-experimentalastawareremapping) to `true`.
+
+We are planning to make this the default remapping mode in the next major. The old `v8-to-istanbul` will be removed completely. Feel free to join discussion at https://github.com/vitest-dev/vitest/issues/7928.
+
+## `watchTriggerPatterns` Option
+
+When you edit a file, Vitest is smart enough to rerun only tests that import this file. Unfortunetly, Vitest static analysis respects only static and dynamic `import` statement. If you are reading a file or start a process, Vitest will ignore the change to that file.
```

Comment:
```suggestion
When you edit a file, Vitest is smart enough to rerun only tests that import this file. Unfortunetly, Vitest static analysis respects only static and dynamic `import` statement. If you are reading a file or start a process, Vitest will ignore the change of that file.
```

Reply:
It's `change to`: https://hinative.com/questions/25498636